protocol/daemons/pricefeed/client/sub_task_runner.go

```diff
@@ -87,8 +91,17 @@ func (s *SubTaskRunnerImpl) StartPriceUpdater(
 		select {
 		case <-ticker.C:
 			err := RunPriceUpdaterTaskLoop(ctx, exchangeToMarketPrices, priceFeedServiceClient, logger)
-			if err != nil {
-				panic(err)
+
+			if err == nil {
```

Comment:
Should something similar happen for the following subtasks too?
* `StartPriceEncoder`
* `StartPriceFetcher`
* `StartMarketParamUpdater`

Why or why not?

Reply:
- StartPriceEncoder
- StartPriceFetcher

No, because I think measuring the contents of the market update already captures more or less the same information, and is much simpler, than instrumenting these subtasks as well.

- StartMarketParamUpdater
I think we made the decision that if a market update fails, we will continue to operate the protocol with the current market configuration on the daemon so that trading can continue for existing markets. (The reasonable IMO assumption here is that market param updates are failing across the network.) Marking unhealthy and panicking on a failure here could bring down the network instead of allowing degraded performance by continuing to support trading for existing markets. The better approach would probably be to continue the network and allow another governance vote to fix any bad parameters in the exchange_config_json.

One possibility to consider is that some daemons could fail and others succeed due to disagreeing app versions, but I think we should catch this by upgrading via governance to enforce versions if there is a breaking change in daemon configuration.

Reply:
Okay, makes sense.

For `StartMarketParamUpdater` do we have clear msgs to indicate this failed? Do we have dashboard + alerting on this?